plugins/woocommerce/includes/admin/settings/class-wc-settings-general.php

```diff
@@ -298,6 +358,14 @@ protected function get_settings_for_default_section() {
 				),
 			);
 
+		// Remove any empty items from settings array.
+		// e.g. The preferred autocomplete provider setting would be empty if <=1 providers are registered.
+		$settings = array_filter(
```

Comment:
Wouldn't it be more performant to remove the empty value for `autocomplete_preferred_provider_setting`? Also, I'm not sure if there is an impact on other settings here being account as empty.